hotness/consumers.py

```diff
@@ -160,17 +127,35 @@ def __init__(self, hub):
 
     def publish(self, topic, msg):
         """
-        Publish a fedmsg message to the specified topic.
+        Publish a feora-messaging message to the specified topic.
+
+        Args:
+            topic (str): Topic to publish to
+            msg (dict): Message to be send
         """
         _log.info("publishing topic %r" % topic)
-        fedmsg.publish(modname="hotness", topic=topic, msg=msg)
+        if self.config["legacy_messaging"]:
+            fedmsg.publish(modname="hotness", topic=topic, msg=msg)
+        else:
+            try:
+                fm_publish(Message(topic=topic, body=msg))
```

Comment:
Actually, fedmsg mangles the topic you give it and prefixes the modname value. To get the equivalent in fedora-messaging, you must use `"hotness.{}".format(topic)` for the topic argument.

Reply:
Ok, didn't knew this.